asset/asset.go

```diff
@@ -1072,7 +1101,12 @@ func NewGKRCustomSubtreeRootRecord(root *chainhash.Hash) tlv.Record {
 //   - [OP_RETURN] is a non-spendable script leaf containing the script
 //     `OP_RETURN`. Its presence ensures that [tweaked_custom_branch] remains
 //     a branch node and cannot be a valid genesis asset ID leaf.
+//
+// TODO(roasbeef): upgrade diagram w/ v1 and v2
```

Comment:
In updating the diagram maybe we can just do:
```
//	                       [tapscript_root]
//	                         /          \
//	[non_spend(<genesis asset ID>)]   [tweaked_custom_branch]
//	                                        /        \
//                                    [non_spend()]    <custom_root_hash>
```

And we can describe `non_spend(...)` in the doc for `NewNonSpendableScriptLeaf`.

`non_spend` would be a nice (Taproot) Miniscript primitive type.

Reply:
Good idea, done. 

I had some trouble with the way `gofmt` seems to want to format comments. At least with my editor set up, I just keeps removing the tabs I add to create some indentation for the new section.